### PR TITLE
sriov: Add a hostdev attach/detach test

### PR DESCRIPTION
**Pull request by Yingshun**
This pr adds:
    VIRT-293236 - Attach-device of hostdev type to guest for
        special scenarios - positive

Signed-off-by: Yingshun Cui <yicui@redhat.com>
**Test result:**
```
  (1/3) type_specific.io-github-autotest-libvirt.sriov.plug_unplug.attach_detach_device_special_situations.hostdev_interface.boot_order: PASS (73.27 s)
 (2/3) type_specific.io-github-autotest-libvirt.sriov.plug_unplug.attach_detach_device_special_situations.hostdev_interface.readonly_mode: PASS (76.88 s)
 (3/3) type_specific.io-github-autotest-libvirt.sriov.plug_unplug.attach_detach_device_special_situations.hostdev_device.unassigned_address: PASS (71.41 s)
```

### CODE REVIEW

**Review by cliping — APPROVED** [review]
LGTM

**Review by vasekhodina — COMMENTED** [review]
I have posted only 2 comments. I think this is a really good code, especially I like the log function calls in setup_test() and run_test().

File: provider/sriov/sriov_base.py [hunk 1]
```diff
@@ -214,6 +216,20 @@ def get_dev_name(self):
...
 
+    def get_rom_file(self):
+        """
+        Get rom file path
+
+        :return: rom file path
+        """
+        cmd = "lspci -nn | awk '/%s/ {print $12}'" % self.vf_pci[5:]
+        rom_vendor_device = process.run(cmd, shell=True)\
+            .stdout_text.strip()[1:-1].replace(':', '') + '.rom'
```
[the remark this answers is not in the document]
Comment (Yingshun) [reply provider/sriov/sriov_base.py]: Sounds good, updated. Thanks!

File: libvirt/tests/src/sriov/plug_unplug/sriov_attach_detach_device_special_situations.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    """
+    def check_vm_iface(vm_session):
+        """
+        Check the VM's interface
+
+        :param vm_session: The session to the VM
+        """
+        # FIXME: If the "IP" command is executed immediately after the interface
+        # is attached, it will be hung.
+        time.sleep(10)
```
[the remark this answers is not in the document]
Comment (Yingshun) [reply libvirt/tests/src/sriov/plug_unplug/sriov_attach_detach_device_special_situations.py:28]: Thanks for your comments! I know it's not a good idea to sleep 10(hard-code) seconds here, but I have no other way for now to make this test stable, so I prefer to setup a "FIXME" keyword and fix it in the future if i got a better way. What do you think?

[the remark this answers is not in the document]
Comment (vasekhodina) [reply libvirt/tests/src/sriov/plug_unplug/sriov_attach_detach_device_special_situations.py:28]: What I meant was if it wouldn't be better to write the warning into the function docstring, remove the time.sleep(10) from the function and move it in front of the function call. But now that I think about it it's probably not worth it as this function is not reused much. Therefore we can afford for the function to be less efficient. I'm approving this PR, even if you won't make changes here.

**Review by vasekhodina — APPROVED** [review]
LGTM

**Review by chloerh — APPROVED** [review]

**Pull request merged**